src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -229,6 +229,7 @@ double SolverInterfaceImpl::initialize()
 {
   PRECICE_TRACE();
   PRECICE_CHECK(_state == State::Constructed, "initialize() may only be called once.");
+  PRECICE_CHECK(_state != State::Finalized, "initialize() cannot be called after finalize().")
```

Comment:
This is already covered by the proceeding check.

Reply:
I would actually suggest to swap the two checks. Then we have more helpful error messages. We would still catch everything by checking for `_state == State::Constructed`, but there are different causes for `_state == State::Constructed` failing:

1. `initialize()` called multiple times
2. `initialize()` called after finalize

I think we should also provide two different error messages.